bark/world/world.cpp

```diff
@@ -70,6 +70,7 @@ void World::PlanAgents(const double& delta_time) {
     if (agent.second->IsValidAtTime(world_time_)) {
       ObservedWorld observed_world = observer_->Observe(
         current_world, agent.first);
+      agent.second->SetSensedWorld(std::make_shared<ObservedWorld>(observed_world));
```

Comment:
Isn't a `SensedWorld` also an `ObservedWorld`?

Reply:
This might cause confusion

Reply:
Later on we may introduce the perceived world which is what a perception component will provide to the agent. The sensed world is then passed to the perception component to generate a perceived world. Therefore, we need to differentiate though it is basically the same type. I do not know if it makes sense to define separate derived types.

Reply:
I think the difference between perceived and sensed world is difficult to grasp.

`SimulatedWorld` and `SensedWorld` would enlarge the gap.

So the `SimulatedWorld` world could model some uncertainties according to a specified (unknown to the agent) distribution and the agent has a `SensedWorld` that it, e.g., could use in a Kalman filter.

Reply:
The questions is do we need an extra type definition or do we just name them differently. I do not understand what you prefer yet.

Reply:
I think we can still rename things and stuff, but there is no way around an agent holding a pointer from its last observed world. If we want to persist the information somewhere its the best place for me.